File: 0162-find-peak-element/0162-find-peak-element.py
class Solution(object):
    def findPeakElement(self, nums):
        if len(nums)==1 or nums[0]>nums[1]:
            return 0
        elif nums[-1]>nums[-2]:
            return len(nums)-1
        for i in range(1,len(nums)-1):
            if nums[i]>nums[i-1] and nums[i]>nums[i+1]:
                return i

        

File: 0162-find-peak-element/0162-find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        int value = nums[0];
        int i=0;
        if(nums.length==1){
            return 0;
        }else if(nums[nums.length-1]>nums[nums.length-2]){
            return nums.length-1;
        }

        while( i<nums.length-1 && nums[i]>=value ){
            value=nums[i];
            i++;
        }
        return i-1;
    }
}